interface/src/app/sample_annotation/annotation.spec.js

```diff
@@ -0,0 +1,68 @@
+describe('sample_annotation', function() {
+  var $httpBackend, $log, SampleMocks;
+
+  beforeEach(module('adage.sampleAnnotation'));
+  beforeEach(module('adage.mocks.sample'));
+  beforeEach(inject(function(_$httpBackend_, _$log_, _SampleMocks_) {
+    $httpBackend = _$httpBackend_;
+    $log = _$log_;
+    SampleMocks = _SampleMocks_;
+  }));
+
+  // Ensure no outstanding expectation or request at the end.
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+    // very useful $log capture information from:
+    // http://www.jvandemo.com/how-to-access-angular-log-debug-messages-from-within-karma/
+    if ($log.debug.logs.length > 0) {
+      console.log($log.debug.logs);
+    }
+  });
+
+  describe('SampleAnnotationCtrl', function() {
+    var SampleAnnotationCtrl, $location, $scope, mockStateParams, $log;
```

Comment:
You are 100% correct about the `$location`, `$scope` and the extra redundant `$log`. I kept the `mockStateParams`, though, since that's how I'm specifying for the unit test what the `samples=` parameter would have been if this were invoked by `ui.router`. This is where we will need to also specify a `node` parameter when that test is ready to go.